src/client/Health/NodeView.jsx

```diff
@@ -25,37 +37,232 @@ const height = 500
 const width = 1240
 
 const NodeView = ({ status, onRestart, onShutdown }) => {
-  const { activeConfig } = React.useContext(ConfigContext)
+  const { setAlert } = React.useContext(AlertContext)
+  const { activeConfig, dispatchNewConfig, apiList, protocolList } = React.useContext(ConfigContext)
   const applications = activeConfig?.north?.applications
   const dataSources = activeConfig?.south?.dataSources
   const engine = activeConfig?.engine
+  const history = useHistory()
+
+  // North Tools *********************
+
+  // create a sortable copy to matain original order in case of sort
+  const sortableApplications = utils.jsonCopy(applications ?? [])
+  // add index for each north application for later use in case of sort
+  sortableApplications?.forEach((application, index) => {
+    application.index = index
+  })
+
+  const getApplicationIndex = ((applicationId) => {
+    const position = sortableApplications.findIndex((application) => application.applicationId === applicationId)
+    if (position === -1) {
+      return position
+    }
+    return sortableApplications[position].index
+  })
+
+  const addApplication = ({ applicationId, api }) => {
+    const applicationIndex = getApplicationIndex(applicationId)
+    if (applicationIndex === -1) {
+      // Adds new application
+      dispatchNewConfig({ type: 'addRow', name: 'north.applications', value: { applicationId, api, enabled: false } })
+    } else {
+      const error = new Error('application already exists')
+      setAlert({ text: error.message, type: 'danger' })
+      throw error
+    }
+  }
+
+  const handleDeleteNorth = (position) => {
+    dispatchNewConfig({ type: 'deleteRow', name: `north.applications.${sortableApplications[position].index}` })
+  }
+
+  const handleDuplicateNorth = (position) => {
+    const application = applications[sortableApplications[position].index]
+    const newName = `${application.applicationId} copy`
+    const countCopies = applications.filter((e) => e.applicationId.startsWith(newName)).length
+    dispatchNewConfig({
+      type: 'addRow',
+      name: 'north.applications',
+      value: {
+        ...application,
+        applicationId: `${newName}${countCopies > 0 ? countCopies + 1 : ''}`,
+        enabled: false,
+      },
+    })
+  }
+
+  const handleEditNorth = (position) => {
+    const application = sortableApplications[position]
+    const link = `/north/${application.applicationId}`
+    history.push({ pathname: link })
+  }
```

Comment:
Since only applicationId is used, you can directly pass applicationId in the arguments. Adapt the method call do pass only the application Id. Do something similar for south